flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/RecordWriter.java

```diff
@@ -72,17 +73,29 @@
 
 	private Counter numBuffersOut = new SimpleCounter();
 
+	/** Default name for teh output flush thread, if no name with a task reference is given. */
+	private static final String DEFAULT_OUTPUT_FLUSH_THREAD_NAME = "OutputFlusher";
+
+	/** The thread that periodically flushes the output, to give an upper latency bound. */
+	private final OutputFlusher outputFlusher;
+
+	/** The exception encountered in the flushing thread. */
+	private Throwable flusherException;
```

Comment:
How this can be not `sychronized`/`volatile`?

Reply:
I talked about it with @NicoK and we have a suspicion that just best effort error checking here is enough. Adding synchronisation on the other hand would add a synchronisation overhead on the critical path (checking for the error happens per each record emission). So maybe it would be best to leave it as it is, but adding a comment that we are fine here with "error being thrown usually/eventually"?

Probably it is a good idea to add a `LOG.error` call to `StreamRecordWriter#notifyFlusherException` to make this error at least always visible.